pkg/serverless/remoteconfig/remoteconfig.go

```diff
@@ -0,0 +1,94 @@
+package remoteconfig
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/comp/remote-config/rctelemetryreporter/rctelemetryreporterimpl"
+	"github.com/DataDog/datadog-agent/pkg/config/model"
+	remoteconfig "github.com/DataDog/datadog-agent/pkg/config/remote/service"
+	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
+	configUtils "github.com/DataDog/datadog-agent/pkg/config/utils"
+	"github.com/DataDog/datadog-agent/pkg/telemetry"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/version"
+	"github.com/aws/aws-sdk-go-v2/aws/arn"
+)
+
+func StartRCService(functionARN string) *remoteconfig.CoreAgentService {
```

Comment:
Largely copied from https://github.com/DataDog/datadog-agent/blob/main/comp/remote-config/rcservice/rcserviceimpl/rcservice.go, have to do this because we don't use [fx](https://github.com/uber-go/fx)

Reply:
There is an on going task to decouple components from Fx. Unfortunately, this is not already done for `comp/remote-config`.
In the meantime, would it be possible to create a new package (for example comp/remote-config/rcservice/rcserviceimpl-nofx) that contains the shared logic between `newRemoteConfigService` and `StartRCService`?

Reply:
That makes sense, I can do that. @timothyalexandersoftware do you have any thoughts from the RC side?

Reply:
@timothyalexandersoftware:  [This document](https://github.com/DataDog/datadog-agent/blob/7.59.x/docs/public/components/creating-components.md) contains all the information about the new component layout.

Reply:
@TalUsvyatsky @ogaca-dd I believe serverless is using Fx 
https://github.com/DataDog/datadog-agent/pull/30381/files#diff-80867e42400e0fcf41760721c1e7bf703a1064d0636bfe72007834f0f21f7475R78-R81

It is a matter of initializing RC and their dependencies, rather than copying. @TalUsvyatsky is this something we tried?

Reply:
We do call the runAgent entry point using fxutil.OneShot, but we don't currently inject any of our dependencies with it (except recently the light noop tagger). I believe that if we use RC components here, we'd have to do quite a bit of additional refactoring, for example with respect to the config module vs. using pkgconfig directly

Reply:
> is that if we use RC components here, we'd have to do quite a bit of additional refactoring of our existing dependencies, for example with respect to the config module vs. using pkgconfig directly

It would be hard for me to answer without first testing it myself. 

I would say try adding the rc component in the `fxutil.OneShot` and see what are the dependencies you need to provide. 

For sure the config component is going to be one of them, which would be a good thiong and you can stop relaying on `pkgconfig` package and use the config component 😄 

I'm happy to assist you @TalUsvyatsky

Reply:
Thanks Gustavo. I gave this a try, and though I got the RC support working, I also saw [unrelated serverless integration tests fail](https://github.com/DataDog/datadog-agent/actions/runs/11788843546/job/32836617474) as well as [a noticeable increase in cold start time](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/703693029). Given that using the fx component has broader implications on performance and correctness of other features, I plan to move forward with the current state of this PR and revisit fx later.